Quadriga/src/main/java/edu/asu/spring/quadriga/web/network/NetworkListController.java

```diff
@@ -15,87 +15,150 @@
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IProjectSecurityChecker projectSecurity;
+
+    @Autowired
+    private IWorkspaceManager workspaceManager;
+
+    @Autowired
+    private IRetrieveProjectManager projectManager;
+
+    private static final Logger logger = LoggerFactory.getLogger(NetworkListController.class);
+
+    /**
+     * This method helps in listing of network belonging to the user in tree
+     * view.
+     * 
+     * @author Lohith Dwaraka
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws JSONException
+     */
+    @RequestMapping(value = "auth/networks", method = RequestMethod.GET)
     public String listNetworks(ModelMap model, Principal principal) throws QuadrigaStorageException, JSONException {
-		IUser user = userManager.getUser(principal.getName());
-		List<INetwork> networkList=networkManager.getNetworkList(user);
-		model.addAttribute("userId", user.getUserName());
-		model.addAttribute("networkList", networkList);
-		return "auth/networks";
-	}
-
-
-	/**
-	 * Get the network displayed on to JSP by passing JSON string
-	 * @author Lohith Dwaraka
-	 * @param networkId
-	 * @param model
-	 * @param principal
-	 * @return
-	 * @throws QuadrigaStorageException
-	 * @throws JAXBException
-	 */
-	@RequestMapping(value = "auth/networks/visualize/{networkId}", method = RequestMethod.GET)
-	public String visualizeNetworks(@PathVariable("networkId") String networkId, ModelMap model, Principal principal) throws QuadrigaStorageException, JAXBException {
-		INetwork network = networkManager.getNetwork(networkId);
-		if(network==null){
-			return "auth/404";
-		}
-		ITransformedNetwork transformedNetwork= transformationManager.getTransformedNetwork(networkId);
-		
-		String nwId = "\""+networkId+"\"";
-		model.addAttribute("networkid",nwId);
-		model.addAttribute("network", network);
-		
-		String json = null;
-		if(transformedNetwork!=null){
-			json = d3Creator.getD3JSON(transformedNetwork.getNodes(), transformedNetwork.getLinks());
-		}
-		model.addAttribute("jsonstring",json);
-
-		return "auth/networks/visualize";
-	}
+        IUser user = userManager.getUser(principal.getName());
+        List<INetwork> networkList = networkManager.getNetworkList(user);
+        model.addAttribute("userId", user.getUserName());
+        model.addAttribute("networkList", networkList);
+        return "auth/networks";
+    }
+
+    /**
+     * Get the network displayed on to JSP by passing JSON string
+     * 
+     * @author Lohith Dwaraka, Chiraag Subramanian
+     * @param networkId
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @RequestMapping(value = "auth/networks/visualize/{networkId}", method = RequestMethod.GET)
+    public String visualizeNetworks(@PathVariable("networkId") String networkId, ModelMap model, Principal principal)
```

Comment:
method should not use the plural of network since it only visualizes one network at a time.